scriptworker/config.py

```diff
@@ -65,24 +65,27 @@ def unfreeze_values(dictionary):
     A recursive function(top-down conversion)
 
     Args:
-        dictionary(frozendict/tuple): the frozendict/tuple to modify in-place.
+        dictionary(frozendict/tuple): the frozendict/tuple.
+
+    Returns:
+        dictionary (dict/list): the unfrozen copy.
     """
-    if isinstance(dictionary, dict):
+    if isinstance(dictionary, (frozendict, dict)):
+        dictionary = dict(deepcopy(dictionary))
         for key, value in dictionary.items():
             if isinstance(value, tuple):
-                dictionary[key] = list(value)
-                unfreeze_values(dictionary[key])
+                dictionary[key] = unfreeze_values(dictionary[key])
```

Comment:
As we now return an unfrozen copy, we can simplify unfreeze_values with list/dict comprehensions:
```py
def unfreeze_values(values):
    """Recursively convert tuples into lists, and frozendicts into dicts.

    Conversion is processed bottom-up.

    Args:
        values (any type): the immutable data structure. Mutable structures are accepted.

    Returns:
        values (dict/list): the unfrozen copy.
    """
    if isinstance(values, (frozendict, dict)):
        return {key: unfreeze_values(value) for key, value in values.items()}

    elif isinstance(values, (list, tuple)):
        return [unfreeze_values(value) for value in values]

    # Nothing to unfreeze.
    return values
```